App/Http/Controllers/RoleController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\Eloquent\RoleRepository;
+use App\Enums\UserPermissionsEnum;
+use App\Models\Role;
+use App\Models\Permission;
+
+class RoleController extends Controller
+{
+
+
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    public function index()
+    {
+        $roleRepository = new RoleRepository();
+        $roles= $roleRepository->getAll();
+
+        return view('roles.index',compact("roles"));
+    }
+
+    public function create()
+    {
+        $permissions = UserPermissionsEnum::getKeys();
+        return view('roles.create',compact("permissions"));
+    }
+
+    public function edit($id)
+    {
+        $roleRepository = new RoleRepository();
+        $role = $roleRepository->find($id);
+        $permissions = UserPermissionsEnum::getKeys();
+        return view('roles.edit',compact("permissions","role"));
+    }
+
+    public function store(Request $request)
+    {
+
+        $this->validate($request, [
+            'name' => ['required','unique:roles']
+        ]);
+        $role = new Role($request->all());
+
+        $permissions =[];
+        $permissionTypes = UserPermissionsEnum::getKeys();
+        foreach($request->all() as $permissionType => $value)
+        {
+            if(in_array($permissionType, $permissionTypes))
+            {
+                $permission = new Permission();
+                $permission->type = $permissionType;
+                $permissions[count($permissions)] = $permission;
```

Comment:
not needed to do this. You can do this
```
$permissions[] = $permission; //this is pushing a new record to the array
```

Reply:
Apply this change on all places you have this logic.

Reply:
done